File: ios/ObjectiveC/ObjectiveC/Beacon/11-Permission/Complete/AuthorizationComplete/ViewController.m
//
//  ViewController.m
//  Authorization
//
//  Created by sarra srairi on 22/03/2016.
//  Copyright © 2016 sarra srairi. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    //register the protocol for did range beacon
    [[ATBeaconManager sharedInstance] registerAdtagBeaconBleLocationDelegate:self];

}
-(void)viewDidDisappear:(BOOL)animated{
    [super viewDidDisappear:animated];
    [[ATBeaconManager sharedInstance] registerAdtagBeaconBleLocationDelegate:nil];

}

-(void)checkBleStatus:(CBCentralManagerState)bleStatus
       locationStatus:(CLAuthorizationStatus)locationStatus{
    
    if(locationStatus==kCLAuthorizationStatusDenied || locationStatus == kCLAuthorizationStatusRestricted){
        _labelStatus.text = NSLocalizedString(@"adtag.warning.Location.status.text", nil);
    }else if (bleStatus== CBCentralManagerStateUnauthorized || bleStatus == CBCentralManagerStatePoweredOff){
        _labelStatus.text = NSLocalizedString(@"adtag.warning.ble.status.text", nil);
    }else if(bleStatus == CBCentralManagerStateUnsupported){
        _labelStatus.text = NSLocalizedString(@"adtag.notsupported.ble.status.text", nil);
    }else{
         _labelStatus.text = NSLocalizedString(@"adtag.ready", nil);
    }
  
    [_labelStatus setNeedsDisplay];
}

@end

File: ios/SWIFT/Beacon/11-Permission/Complete/Authorization/ViewController.swift
//
//  ViewController.swift
//  Authorization
//
//  Created by sarra srairi on 10/08/2016.
//  Copyright © 2016 R&D connecthings. All rights reserved.
//

import UIKit
import ATAnalytics
import ATLocationBeacon
import AVFoundation


class ViewController: UIViewController,ATBeaconBleLocationStatusDelegate {

    @IBOutlet weak var labelStatus: UILabel!
    
  
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear()
        ATBeaconManager.sharedInstance().registerAdtagBeaconBleLocationDelegate(self)
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    
    override func viewDidDisappear(animated: Bool) {
        super.viewDidDisappear(animated)
        ATBeaconManager.sharedInstance().registerAdtagBeaconBleLocationDelegate(nil)
    }
    
    
    func checkBleStatus(bleStatus: CBCentralManagerState, locationStatus: CLAuthorizationStatus) {
        
        if locationStatus == CLAuthorizationStatus.Denied || locationStatus == CLAuthorizationStatus.Restricted {
            self.labelStatus.text = "Please authorize the application to access to your location"
        }
        else if bleStatus == .Unauthorized || bleStatus == .PoweredOff {
            self.labelStatus.text = "Please activate bluetooth"
        }
        else if bleStatus == .Unsupported {
            self.labelStatus.text = "The bluetooth is not supported on your phone" 
        }
        else {
            self.labelStatus.text = "It's ready for beacon detection"
        }
        
        labelStatus.setNeedsDisplay()
    }
}

